Mathlib/Data/Int/SuccPred.lean

```diff
@@ -36,46 +40,43 @@ theorem succ_eq_succ : Order.succ = succ :=
 theorem pred_eq_pred : Order.pred = pred :=
   rfl
 
+@[deprecated Order.one_le_iff_pos (since := "2024-09-04")]
 theorem pos_iff_one_le {a : ℤ} : 0 < a ↔ 1 ≤ a :=
   Order.succ_le_iff.symm
 
-theorem succ_iterate (a : ℤ) : ∀ n, succ^[n] a = a + n
-  | 0 => (add_zero a).symm
-  | n + 1 => by
-    rw [Function.iterate_succ', Int.ofNat_succ, ← add_assoc]
-    exact congr_arg _ (succ_iterate a n)
+protected theorem succ_iterate (a : ℤ) : ∀ n, succ^[n] a = a + n :=
```

Comment:
Should this one be deprecated as well?

Reply:
This theorem can be used to immediately prove both `Nat.succ_iterate` and `Int.succ_iterate`, which I think is pretty cool. It could potentially be used for stuff on archimedean orders, and it follows our design of using arithmetic instead of `succ`.

Reply:
Ah, misread. Well, `Nat.succ_iterate` isn't actually syntactically equal to `Order.succ_iterate`, since one uses `Order.succ` and the other uses `Nat.succ`.

Reply:
I didn't know that was the plan! It makes a lot of sense though, it's not even a constructor like `Nat.succ` is. Will deprecate these both.

`Order.succ_iterate` and `Order.pred_iterate` can stay though, right?

Reply:
Yes, the plan is to make them replace the above lemma 😉